pkg/admission/validator/cache/shoot_test.go

```diff
@@ -227,6 +227,26 @@ var _ = Describe("Shoot validator", func() {
 					"Detail":   Equal("field is immutable"),
 				}))))
 			})
+
+			It("should exit earlier when no semantic change in providerConfig is detected", func() {
+				shoot.Spec.Extensions[0].ProviderConfig = &runtime.RawExtension{
+					Raw: encode(&v1alpha3.RegistryConfig{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: v1alpha3.SchemeGroupVersion.String(),
+						},
+						Caches: []v1alpha3.RegistryCache{},
+					}),
+				}
+				oldShoot.Spec.Extensions[0].ProviderConfig = &runtime.RawExtension{
+					Raw: encode(&v1alpha3.RegistryConfig{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: v1alpha3.SchemeGroupVersion.String(),
+						},
+						Caches: nil,
+					}),
+				}
+				Expect(shootValidator.Validate(ctx, shoot, oldShoot)).To(Succeed())
+			})
```

Comment:
We could use more real world test data:
```suggestion
			It("should exit earlier when no semantic change in providerConfig is detected", func() {
				shoot.Spec.Extensions[0].ProviderConfig = &runtime.RawExtension{
					Raw: encode(&v1alpha3.RegistryConfig{
						TypeMeta: metav1.TypeMeta{
							APIVersion: v1alpha3.SchemeGroupVersion.String(),
							Kind:       "RegistryConfig",
						},
						Caches: []v1alpha3.RegistryCache{
							{
								Upstream: "https://registry.example.com", // invalid upstream
							},
						},
					}),
				}
				oldShoot.Spec.Extensions[0].ProviderConfig = &runtime.RawExtension{
					Raw: encode(&v1alpha3.RegistryConfig{
						TypeMeta: metav1.TypeMeta{
							APIVersion: v1alpha3.SchemeGroupVersion.String(),
							Kind:       "RegistryConfig",
						},
						Caches: []v1alpha3.RegistryCache{
							{
								Upstream: "https://registry.example.com", // invalid upstream
								GarbageCollection: &v1alpha3.GarbageCollection{
									TTL: v1alpha3.DefaultTTL,
								},
							},
						},
					}),
				}
```